code/modules/food_and_drinks/kitchen_machinery/microwave.dm

```diff
@@ -256,6 +256,11 @@
 					dirty++
 			qdel(F)
 
+		for(var/obj/item/weapon/dice/D in contents)
```

Comment:
it would be worth considering adding a
```DM
/obj/proc/microwave_act()
```
and moving all the stuff such as cooking and dice rigging into their respective classes and overriding microwave_act()